### DISCUSSION TITLE
Automatic Proxmox Updates :)

### DISCUSSION THREAD

**Discussion by jascdk** [tteck/Proxmox#3338] [Ideas]
Hi :)

I hate update my Proxmox manually all the time. Would it be a good idea to make a script like this?

```
#!/bin/bash`
 
# Use full paths for commands to ensure they are found by cron
/usr/bin/apt-get update -yy
/usr/bin/apt-get dist-upgrade -yy
 
# The following packages were automatically installed and are no longer required:
# Use 'apt autoremove' to remove them.
/usr/bin/apt-get autoremove -yy
 
# Gracefully shutdown all running VMs
echo "Shutting down running VMs..."
for vmid in $(/usr/sbin/qm list | awk '$3 == "running" {print $1}')
do
    echo "Shutting down VM $vmid..."
    /usr/sbin/qm shutdown $vmid
done
 
# Wait for VMs to shut down
sleep 30
 
# Initiate system reboot
echo "Initiating system reboot at $(date)"
/usr/sbin/reboot

```

I made a file called "update" with the editor and put the above code inside.

I run it with a crontab

```

0 2 * * * /root/update

```

Would it be possible to put this script or make a choose function on the during the post install of proxmox?

All the best

Jacob

**Comment by Mellowlynx** [comment]
If I recall correctly, tteck has not made a script for this because, for obvious reasons, it's not recommended to just update a production environment without testing it. I entirely agree with that, and test new PVE updates for up to about two weeks (depending on the changes) before running my update playbook manually. It will make an extra backup of all the LXC's and VM's, update all the LXC nodes and reboot them when a reboot is needed.

This way, there is a match lower change that your production PVE host(s) will experience problems after an update that has some kind of bug. That may or may not be easy to troubleshoot and fix.

So yes, you can do it your way, (happy to see `dist-upgrade`, allot use `upgrade` :disappointed: ) but keep in mind, if something goes wrong, it can go wrong bad, and everything will be down. Be prepared for that and have a plan.
